pkg/connector/client/salesforce.go

```diff
@@ -179,14 +181,28 @@ func getIsActive(record simpleforce.SObject) (bool, error) {
 	case bool:
 		return v, nil
 	case string:
-		return v == "true", nil
+		return v == trueConst, nil
 	case int, float64:
 		return v == 1, nil
 	default:
 		return false, fmt.Errorf("salesforce-connector: unexpected is active type, %s", value)
 	}
 }
 
+func getBoolField(record simpleforce.SObject, field string) (bool, error) {
+	value := record.InterfaceField(field)
+	switch v := value.(type) {
+	case bool:
+		return v, nil
+	case string:
+		return v == trueConst, nil
+	case int, float64:
+		return v == 1, nil
```

Comment:
is there any chance that v can be `2` or something larger than `1`?

Reply:
Yes, since the `field` could be any field, the function caller could send any field on `simpleforce.SObject`

Reply:
> Yes, since the `field` could be any field, the function caller could send any field on `simpleforce.SObject`

I mean, what if `v` is `2`? then `2 == 1` would return `false`, won't it? 👀

Reply:
Yes, it's depends of the caller, since any bool it's a "true" response on salesforce

Reply:
then... maybe is better to use `v > 0` isn't? 👀

Reply:
I don't think so, since it's a bool comparer, but this issue looks like a design choice. Like how js/python handles this scenario ex: float == true.

Reply:
> I don't think so, since it's a bool comparer, but this issue looks like a design choice. Like how js/python handles this scenario ex: float == true.

I mean, I don't know what can `v` be if it was an integer if the source of this data always use `1` to signal `true` then everything should be okay.